static/app/components/searchSyntax/parser.tsx

```diff
@@ -955,41 +937,173 @@ function numeric(input: string) {
+): {value: number} {
+  let number = numeric(input);
+
+  switch (unit) {
+    case 'ms':
+      break;
+    case 's':
+      number *= 1e3;
+      break;
+    case 'min':
+    case 'm':
+      number *= 1e3 * 60;
+      break;
+    case 'hr':
+    case 'h':
+      number *= 1e3 * 60 * 60;
+      break;
+    case 'day':
+    case 'd':
+      number *= 1e3 * 60 * 60 * 24;
+      break;
+    case 'wk':
+    case 'w':
+      number *= 1e3 * 60 * 60 * 24 * 7;
+      break;
+    default:
+      throw new Error('Invalid unit');
+  }
+
+  return {
+    value: number,
+  };
 }
-function parseNumber(input: string, unit: 'k' | 'm' | 'b' | 'K' | 'M' | 'B') {
-  const number = numeric(input);
+function parseNumber(
+  input: string,
+  unit: 'k' | 'm' | 'b' | 'K' | 'M' | 'B'
+): {value: number} {
+  let number = numeric(input);
 
   switch (unit) {
     case 'K':
     case 'k':
-      return number * 1e3;
+      number = number * 1e3;
+      break;
     case 'M':
     case 'm':
-      return number * 1e6;
+      number = number * 1e6;
+      break;
     case 'B':
     case 'b':
-      return number * 1e9;
+      number = number * 1e9;
+      break;
     case null:
     case undefined:
-      return number;
+      break;
     default:
       throw new Error('Invalid unit');
   }
+
+  return {value: number};
 }
-function parseSize(input: string, _unit: string) {
-  return numeric(input);
+function parseSize(
+  input: string,
+  // warning: size units are case insensitive, this type is incomplete
+  unit:
+    | 'bit'
+    | 'nb'
+    | 'bytes'
+    | 'kb'
+    | 'mb'
+    | 'gb'
+    | 'tb'
+    | 'pb'
+    | 'eb'
+    | 'zb'
+    | 'yb'
+    | 'kib'
+    | 'mib'
+    | 'gib'
+    | 'tib'
+    | 'pib'
+    | 'eib'
+    | 'zib'
+    | 'yib'
+): {value: number} {
+  if (!unit) {
+    unit = 'bytes';
+  }
+
+  let number = numeric(input);
+
+  switch (unit.toLowerCase()) {
+    case 'bit':
+      number /= 8;
+      break;
+    case 'nb':
+      number /= 2;
+      break;
+    case 'bytes':
+      break;
+    case 'kb':
+      number *= 1000;
+      break;
+    case 'mb':
+      number *= 1000 ** 2;
+      break;
+    case 'gb':
+      number *= 1000 ** 3;
+      break;
+    case 'tb':
+      number *= 1000 ** 4;
+      break;
+    case 'pb':
+      number *= 1000 ** 5;
+      break;
+    case 'eb':
+      number *= 1000 ** 6;
+      break;
+    case 'zb':
+      number *= 1000 ** 7;
+      break;
+    case 'yb':
+      number *= 1000 ** 8;
+      break;
+    case 'kib':
+      number *= 1024;
+      break;
+    case 'mib':
+      number *= 1024 ** 2;
+      break;
+    case 'gib':
+      number *= 1024 ** 3;
+      break;
+    case 'tib':
+      number *= 1024 ** 4;
+      break;
+    case 'pib':
+      number *= 1024 ** 5;
+      break;
+    case 'eib':
+      number *= 1024 ** 6;
+      break;
+    case 'zib':
+      number *= 1024 ** 7;
+      break;
+    case 'yib':
+      number *= 1024 ** 8;
+      break;
```

Comment:
more of a nit, but you could pretty easily write this with a lookup table no?

```tsx
const factor = {
  kb: 1000,
  mb: 1000 ** 2,
  ...
  kib: 1024,
  mib: 1024 ** 2,
  ...
} 

number *= factor[unit.toLowerCase()]
```

Reply:
Yeah, I know. I copied as the exact same code from the backend so that the flow is the same. We could update it on both ends tbh